### PR TITLE
[REEF-1020] Improve OSUtils to support Windows and Mac

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
JIRA:
  [REEF-1020](https://issues.apache.org/jira/browse/REEF-1020)

Pull request:
  This closes #

### CODE REVIEW

File: lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java [hunk 1]
```diff
@@ -47,11 +47,12 @@ public static boolean isWindows() {
    * @return true, if the JVM is running on Linux. false, otherwise
    */
   public static boolean isLinux() {
-    return System.getProperty("os.name").toLowerCase().contains("linux");
+    final String os = System.getProperty("os.name").toLowerCase();
+    return os.contains("linux") || os.contains("mac");
```
Comment (markusweimer) [inline lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java]: This is dangerous. Shouldn't we better add a method `isMac()`? Also, we can have `isUnix()` to indicate that the OS is either Mac OS or Linux.

File: lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureDuringAlarmDriver.java [hunk 2]
```diff
@@ -70,6 +71,12 @@ public void onNext(final FailedEvaluator failedEvaluator) {
     }
   }
 
+  final class ActiveContextHandler implements EventHandler<ActiveContext> {
```
Comment (markusweimer) [inline lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureDuringAlarmDriver.java]: The changes in this file seem orthogonal to the goals of the PR?

File: lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureTest.java [hunk 3]
```diff
@@ -61,6 +61,7 @@ public void testEvaluatorFailureByAlarmHandler() throws InjectionException {
             EnvironmentUtils.getClassLocation(EvaluatorFailureDuringAlarmDriver.class))
         .set(DriverConfiguration.DRIVER_IDENTIFIER, "TEST_EvaluatorFailureTest")
         .set(DriverConfiguration.ON_DRIVER_STARTED, OnDriverStartedAllocateOne.class)
+        .set(DriverConfiguration.ON_CONTEXT_ACTIVE, EvaluatorFailureDuringAlarmDriver.ActiveContextHandler.class)
```
Comment (markusweimer) [inline lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureTest.java]: The changes in this file seem orthogonal to the goals of the PR?

**Comment by markusweimer** [comment]
@dongjoon-hyun I've done a pass.

File: lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java (hunk 1 above)
Comment (dongjoon-hyun) [inline lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java]: I see. No problem.

File: lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureDuringAlarmDriver.java (hunk 2 above)
Comment (dongjoon-hyun) [inline lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureDuringAlarmDriver.java]: Actually, the test on Windows fails after implementing OSUtils fully. The reason is `DefaultContextActiveHandler` really works immediately as I wrote in JIRA issue.
```
Add No-Op `ActiveContextHandler` to EvaluatorFailureDuringAlarmDriver. `DefaultContextActiveHandler` suicides by closing itself.
```

File: lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureTest.java (hunk 3 above)
Comment (dongjoon-hyun) [inline lang/java/reef-tests/src/test/java/org/apache/reef/tests/evaluatorfailure/EvaluatorFailureTest.java]: I fixed this testsuite in the same reason above.

**Comment by dongjoon-hyun** [comment]
Thank you, @markusweimer . For the changes on testsuites, I made that changes to pass the Unit tests.

**Comment by dongjoon-hyun** [comment]
I changed the code according the comments and also rebased to the master.

**Comment by markusweimer** [comment]
Thanks! I will test and merge.

File: lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java [hunk 4]
```diff
@@ -64,31 +82,50 @@ public static long getPID() {
...
         return result;
-      } catch (final IOException e) {
+      } catch (final Exception e) {
+        LOG.log(Level.SEVERE, "Unable to determine PID", e);
+        return -1;
+      }
+    } else if (isWindows()) {
+      try {
+        final Process process = new ProcessBuilder()
+            .command("powershell.exe", "-Command", "wmic process where processid=$pid get parentprocessid")
```
Comment (markusweimer) [inline lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java]: During testing, I noticed that this fails on my machine because my profile prints out stuff before the printout this looks for. I replaced it 

```
.command("powershell.exe", "-NoProfile", "-Command", "wmic process where processid=$pid get parentprocessid")
```

which fixed the issue. I will merge it this way.

Comment (dongjoon-hyun) [inline lang/java/reef-common/src/main/java/org/apache/reef/util/OSUtils.java]: Oh, thank you!
